packages/angular_devkit/core/src/json/schema/visitor.ts

```diff
@@ -110,13 +142,18 @@ function _visitJsonRecursive(
  *
  * @param {JsonValue} json The Json value to visit.
  * @param {JsonVisitor} visitor A function that will be called on every items.
+ * @param {JsonObject} schema A JSON schema to pass through to the visitor (where possible).
+ * @param refResolver a function to resolve references in the schema.
  * @returns {Observable< | undefined>} The observable of the new root, if the root changed.
  */
-export function visitJson(
+export function visitJson<ContextT>(
```

Comment:
Is there any guarantee that the JsonValue is not cyclical? If you want to accept recursive JsonValues, you'll need to transform this into a depth first with open/closed lists to not be stuck in loops.

Some of the AJV operations can end up having references instead of values.

Reply:
There's none. We cannot really go through that either (it would be too memory and time consuming). But at that point isn't it the same as an infinite loop on the part of the API user?